### ISSUE TITLE
NUnit2026 shouldn't fire when CustomEqualityComparer is used

### ISSUE THREAD

**Issue by mikkelbu** [nunit/nunit.analyzers#332]
The assertion below raises NUnit2026 `The 'Does.Contain' constraint cannot be used with actual argument of type 'string[]' and expected argument of type 'int'`, but there is no problem as we supply a custom comparer function. So if a custom comparer function is supplied to the constraint then we must not raise NUnit2026.

```csharp
[Test]
public void TestA()
{
    var actual = new[] { "1", "2", "3" };
    Func<string, int, bool> func = (a, b) => true;
    #pragma warning disable NUnit2026 // Wrong actual type used with the SomeItemsConstraint with EqualConstraint
    Assert.That(actual, Does.Contain(1).Using(func));
    #pragma warning restore NUnit2026 // Wrong actual type used with the SomeItemsConstraint with EqualConstraint
}
```

**Comment by manfred-brands** [comment]
@mikkelbu I will look into this.
